test/unit/BCoWPool/BCoWPoolBase.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import {BPoolBase} from '../BPool/BPoolBase.sol';
+import {BCoWConst} from 'contracts/BCoWConst.sol';
+import {BNum} from 'contracts/BNum.sol';
+
+import {ISettlement} from 'interfaces/ISettlement.sol';
+import {MockBCoWPool} from 'test/manual-smock/MockBCoWPool.sol';
+
+contract BCoWPoolBase is BPoolBase, BCoWConst, BNum {
+  bytes32 public appData = bytes32('appData');
+  address public cowSolutionSettler = makeAddr('cowSolutionSettler');
+  bytes32 public domainSeparator = bytes32(bytes2(0xf00b));
+  address public vaultRelayer = makeAddr('vaultRelayer');
+  address public tokenIn;
+  address public tokenOut;
+  MockBCoWPool bCoWPool;
+
+  function setUp() public virtual override {
+    super.setUp();
+    tokenIn = tokens[0];
+    tokenOut = tokens[1];
```

Comment:
doesn't this happen in setup?

Reply:
no, tokens are not given a name in `BPoolBase`, and the indices do not give a good sense of which is the input and which the output. This could happen in the derived contract tho, since it's not relevant for *every* tested method, wdyt?

Reply:
i think is ok, tokenIn and Out may not represent all possible cases, so 👌  for me